ruby/vampires.rb

```diff
@@ -4,45 +4,70 @@
 sign_up = true
 result = "Result inconclusive."
 
-puts "What is your name?"
-name = gets.chomp
-name = name.downcase
-if (name == "drake cula") || (name == "tu fang")
-		real_name = false
-end
+puts "How many employees will be processed?"
+employees = gets.chomp
+x = 1
+until x > employees.to_i
 
-puts "How old are you? What year were you born?"
-age = gets.chomp
-year = gets.chomp
-if (2016 - year.to_i) != age.to_i
-	    real_age = false
-end
+	puts "What is your name?"
+	name = gets.chomp
+	name = name.downcase
+	if ( ( name == "drake cula" ) || ( name == "tu fang" ) )
+			real_name = false
+	end
 
-puts "Our company cafeteria serves garlic bread. Should we order some for you?(y/n)"
-order_bread = gets.chomp
-if (order_bread == "n")
-        garlic_bread = false
-end
+	puts "How old are you? What year were you born?"
+	age = gets.chomp
+	year = gets.chomp
+	if ( ( 2016 - year.to_i ) != age.to_i )
+		    real_age = false
+	end
 
-puts "Would you like to enroll in the company's health insurance?(y/n)"
-insurance = gets.chomp
-if (insurance == "n")
-        sign_up = false
-end
+	puts "Our company cafeteria serves garlic bread. Should we order some for you?(y/n)"
+	order_bread = gets.chomp
+	if ( order_bread == "n" )
+	        garlic_bread = false
+	end
 
+	puts "Would you like to enroll in the company's health insurance?(y/n)"
+	insurance = gets.chomp
+	if ( insurance == "n" )
+	        sign_up = false
+	end
 
-if real_age && ( garlic_bread || sign_up )
-	result = "Probably not a vampire."
-end
-if !real_age && ( !garlic_bread || !sign_up )
-	result = "Probably a vampire."
-end
-if !real_age && !garlic_bread && !sign_up
-	result = "Almost certainly a vampire."
-end
-if !real_name
-	result = "Definitely a vampire."
-end
+	puts "Do you have any allergies? Please list them one at a time. Type 'done' when you are finished."
+	
+		allergies = gets.chomp
+		allergies = allergies.downcase
+	until ( allergies == "done" || allergies == "sunshine" )
+		puts "Any other allergies?"
+		allergies = gets.chomp
+		allergies = allergies.downcase
+	end
 
+	if ( real_age && ( garlic_bread || sign_up ) )
+		result = "Probably not a vampire."
+	end
+
+	if ( !real_age && ( !garlic_bread || !sign_up ) )
+		result = "Probably a vampire."
+	end
+
+	if ( !real_age && !garlic_bread && !sign_up )
+		result = "Almost certainly a vampire."
+	end
+
+	if ( !real_name )

```

Comment:
I think this should be your first check. If their name is Drac or Fang then we know we have a vampire. You can nest the rest of the checks under an else following this check. if vamp == true then no reason to check anything else.

The below is an example of handling the control flow in a more efficient computational pattern. Also, it is important to practice nesting conditions as many control flows will require you to do so.

```
if name == vamp
      vampire = true
else
      if other conditions
      elsif other conditions
      end
end
```